packages/api/db/migration/20240116175706_create_animal_tables.js

```diff
@@ -0,0 +1,206 @@
+/*
+ *  Copyright (c) 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+export const up = async function (knex) {
+  await knex.schema.createTable('animal_sex', (table) => {
+    table.increments('id').primary();
+    table.string('key').notNullable();
+  });
+
+  await knex.schema.createTable('animal_identifier_color', (table) => {
+    table.increments('id').primary();
+    table.string('key').notNullable();
+  });
+
+  await knex.schema.createTable('animal_identifier_placement', (table) => {
+    table.increments('id').primary();
+    table.integer('default_type_id').references('id').inTable('default_animal_type').notNullable();
+    table.string('key').notNullable();
+  });
+
+  await knex.schema.createTable('animal_origin', (table) => {
+    table.increments('id').primary();
+    table.string('key').notNullable();
+  });
+
+  await knex.schema.createTable('animal', (table) => {
+    table.increments('id').primary();
+    table.uuid('farm_id').notNullable().references('farm_id').inTable('farm');
+    table.integer('default_breed_id').references('id').inTable('default_animal_breed');
+    table.integer('custom_breed_id').references('id').inTable('custom_animal_breed');
+    table.integer('sex_id').references('id').inTable('animal_sex');
+    table.string('name');
+    table.dateTime('birth_date');
+    table.string('identifier');
+    table.integer('identifier_color_id').references('id').inTable('animal_identifier_color');
+    table
+      .integer('identifier_placement_id')
+      .references('id')
+      .inTable('animal_identifier_placement');
+    table.integer('origin_id').references('id').inTable('animal_origin');
+    table.string('dam');
+    table.string('sire');
+    table.dateTime('brought_in_date');
+    table.dateTime('weaning_date');
+    table.text('notes');
```

Comment:
I used `text` because it doesn't have a set character limit, we could also go with `string` but I think we'd probably need to increase the 256 character limit in that case depending on what we want that notes field to look like in the UI